src/OrleansAzureUtils/Providers/Storage/AzureTableStorage.cs

```diff
@@ -176,10 +177,20 @@ public Task Close()
                 await tableDataManager.Write(record);
                 grainState.ETag = record.ETag;
             }
+            catch (StorageException exc)
+            {
+                Log.Error((int)AzureProviderErrorCode.AzureTableProvider_WriteError,
+                    $"Error Writing: GrainType={grainType} Grainid={grainReference} ETag={grainState.ETag} to Table={tableName} Exception={exc.Message}", exc);
+                if (exc.IsUpdateConditionNotSatisfiedError())
+                {
+                    throw new InconsistentStateException("Conditions not satisfied", exc);
```

Comment:
Since users will see this exception, maybe the message should be more descriptive, like "Conflict while writing grain state: precondition check failed. This likely indicates a duplicate activation." - I don't know. We could also consider including some context, like what's in the log message.